### ISSUE TITLE
[BLOCKED] Sometimes Partial Data is Created after clicking "Find Closest Churches"

### ISSUE THREAD

**Issue by Julian88Tex** [lovinghouston/lwcgoogleplacesbutton#44]
Steps to Repo:

1. Delete "Yehoshua Missionary Baptist Church": https://lovinghouston--partial.lightning.force.com/lightning/r/Account/0010r00000KS6nEAAT/view
2. Navigate to PARKER INT: https://lovinghouston--partial.lightning.force.com/lightning/r/Account/0011N00001qzT3vQAE/view
3. Click "Find Closest Churches"

Result:

- "20 closest Churches have already been found and created" shows
- "Yehoshua Missionary Baptist Church" is created
- No CSP records are created for the existing church

Expected Result:

- Results display of any existing churches  & any created churches
- CSP records are created for each

**Comment by Julian88Tex** [comment]
@sungysang this one may be tricking I'm guessing since it deals with existing and new records... let me know if that makes it too challenging to do in a reasonable amount of time. At the very least I'd prefer data to not be created if we aren't letting the client know. If there is an unknown error that occurs data should not be created at all and an error should display saying there was an unknown error.

**Comment by sungysang** [comment]
@Julian88Tex A little hesitant on this one just because of how much grief it caused C4SF ... do we want existing data to be overwritten? I could potentially check if the fields are empty first before adding to them. But for now, I think Address and name should be left alone

**Comment by Julian88Tex** [comment]
> @Julian88Tex A little hesitant on this one just because of how much grief it caused C4SF ... do we want existing data to be overwritten? I could potentially check if the fields are empty first before adding to them. But for now, I think Address and name should be left alone

Good point and no wouldn't want data overwritten. Sorry for the confusion tone clear by partial I meant accounts being created with CSPs or vise versa. Nothing to do with number of fields filled in.

**Comment by sungysang** [comment]
@Julian88Tex Sorry could you clarify? Are there new Account records that don't have information filled in clearly? Or does this only affect Accounts that have been created before the button was clicked?

If talking about existing Accounts (created before the button), I could add field information that aren't address and Name and only if they are empty. Would that be a good alternative to not filling in anything at the moment?

**Comment by Julian88Tex** [comment]
@sungysang as far as I can tell the button is creating an account correctly but not creating a CSP for some reason. Theoretically there is an error not being surfaced that I haven't been able to identify.

**Comment by Julian88Tex** [comment]
I haven't looked at any fields being filled in or not filled in for this ticket. If there is an issue like that I'd create a new ticket for it. This is l a our accounts being created without CSPs.

**Comment by sungysang** [comment]
ok will look into it when I have a chance

**Comment by Julian88Tex** [comment]
@sungysang looked at logs a bit more and am seeing that flow error coming up. I tried deactivating it and then running again but CSP records still didn't get created. When I looked in the logs I saw this error:
System.DmlException: Insert failed. First exception on row 23; first error: FIELD_FILTER_VALIDATION_EXCEPTION, Value does not exist or does not match filter criteria (type=school district): [School_District_Lookup__c]

Having a hard time determine if it's the real culprit. Also seeing a lot of DLRS running so that may be related.

All that being said, I suspect the thing to "fix" here is just adding a try and catch anytime there DML and surfacing that to the button message so it's clear if something broke and also not performing any DML if one DML fails.... if that's possible.

**Comment by sungysang** [comment]
@Julian88Tex It's weird that the SD lookup field is showing an error because I don't talk to this field at all and it works fine for other records

**Comment by sungysang** [comment]
Disabling the lookup field filter allows CSP records to be created. As a workaround, this can be disabled before the button is pressed for now

**Comment by sungysang** [comment]
https://lovinghouston--partial.lightning.force.com/lightning/setup/ObjectManager/Account/FieldsAndRelationships/00N1N00000EwRVR/view

**Comment by sungysang** [comment]
@Julian88Tex should this be closed since disabling the lookup filter corrects this? Or are you keeping open until you address turning off the filter in LH's production with them

**Comment by Julian88Tex** [comment]
> @Julian88Tex should this be closed since disabling the lookup filter corrects this? Or are you keeping open until you address turning off the filter in LH's production with them

The latter :-) @sungysang

**Comment by Julian88Tex** [comment]
@sungysang just emailed Melissa to see if she's okay with us removing the lookup filter.

In [production](https://lovinghouston.lightning.force.com/lightning/setup/ObjectManager/Account/FieldsAndRelationships/00N1N00000EwRVR/view), it appears this field isn't being used very often. [Report](https://lovinghouston.lightning.force.com/lightning/r/Report/00O3l000005MEKREA4/view?queryScope=userFolders) showing it's only filled in on 90 accounts.

**Comment by Julian88Tex** [comment]
Got approval form LH to remove Lookup filter. I changed this ticket to Blocked since we will do it once we are ready to move code from Sandbox to Production.
